### DISCUSSION TITLE
How to fix `ChainMismatchError`?

### DISCUSSION THREAD

**Discussion by benesjan** [wevm/viem#413] [Question] [answered]
When I run the test test bellow:
<img width="1612" alt="image" src="https://user-images.githubusercontent.com/13470840/234326751-0b3b85e5-64d9-4dc7-ae2b-74ebf661b1f5.png">

I get the following error:

```
    Cause:
    ChainMismatchError: The current chain of the wallet (id: 31337) does not match the target chain for the transaction (id: 1337 – Localhost).

    Current Chain ID:  31337
    Expected Chain ID: 1337 – Localhost
```

I assume it's caused by the types being set incorrectly. If that's the case how do I define the type of the `deployRollupContract` function that the chain is correctly derived?

`export const deployRollupContract = async (walletClient: **WalletClient<HttpTransport, Chain, Account>**) => {`

I use current newest version of Viem (0.3.11).
Thank you

**Comment by jxom** [comment, marked as answer]
You'll probably need to amend the `id` of `localhost`.

```tsx
const testChain = {
  ...localhost,
  id: 1337
} as const
```

**Reply by benesjan** [reply]
Anvil has default id `31337` so when I used the following code:

```
    const anvil = {
      ...localhost,
      id: 31337,
    } as const

    const account = mnemonicToAccount(MNEMONIC);
    walletClient = createWalletClient({
      account,
      chain: anvil,
      transport: http(config.rpcUrl),
    });

    const hash = await walletClient.deployContract({
      abi: RollupAbi,
      bytecode: RollupBytecode,
    });
```

everything worked.

Thank you
